src/seq/nucleotide.jl

```diff
@@ -97,27 +97,27 @@ end
 
 Test if nucleotide is surely a purine.
 """
-function ispurine(nt::Nucleotide)
+@inline function ispurine(nt::Nucleotide)
     bits = reinterpret(UInt8, nt)
-    return bits != 0 && (bits & 0b1010) == 0
+    return (bits != 0) & ((bits & 0b1010) == 0)
```

Comment:
I haven't benchmarked, but I inspected the llvm instructions generated, and without it there's no instructions to use the simd capabilities of the processor.